Android: Circular Icon in notification

gjerlow opened the pull request:
Added `getCircleBitmap()` method in _Utils.java_ and added a check for the `icon_as_circle` parameter in _TiFirebaseMessagingService.java_

There could be done more to smoothen the border of the circular Bitmap as it might appear a bit pixelated for some images: [smooth canvas shapes with anti-aliasing](https://medium.com/@ali.muzaffar/android-why-your-canvas-shapes-arent-smooth-aa2a3f450eb5)

hansemannn commented on android/src/firebase/cloudmessaging/TiFirebaseMessagingService.java:
```diff
@@ -234,6 +234,17 @@ private Boolean showNotification(RemoteMessage remoteMessage)
 			try {
 				Bitmap icon = this.getBitmapFromURL(params.get("icon"));
+
+				if(params.get("icon_as_circle") != null && params.get("icon_as_circle") != "") {
```
Please use `optBoolean` to have a boolean value directly.

hansemannn replied:
Also, use camelCase for property names, e.g. `iconAsCircle`

gjerlow replied:
Thanks for reviewing. I'll get on it tomorrow

gjerlow replied:
@hansemannn changing the property name to camelCase will differentiate it from e.g. `fcm_data`, `big_text_summary` and `force_show_in_foreground`. Should I continue with the renaming?

hansemannn replied:
Oh, I didn't know that. Looks like those are delivered via JSON anyway right? Then it's fine. But please add a minimal documentation around this property to the readme, so others now it's available.

m1ga commented:
Out of curiosity: What is the benefit of this PR 😃? Why not add a custom round icon in the first place instead of making it round on-the-fly? Or do newer phones have round push icons?

gjerlow commented:
The purpose would be to provide an alternative for round push icons as e.g. avatars for messages from users. Yes, you could make the round icons by default in the app, but we use square icons/avatars in our app and set borderRadius to ImageViews for displaying round images. This way, UI can be changed from round to square without updating images in our DB. 

We made this PR on a fork for our own usage, and it made notification customization easier at least for us. Perhaps most apps have round images by default, but we dont.

![android_notification](https://user-images.githubusercontent.com/8874119/58470044-24b1cc00-8141-11e9-8408-8e81307d04a1.png)

m1ga commented:
Ah ok, so it should be `image as circle`! The Icon is the red Gmail icon in the top left corner. That's why I was confused.

gjerlow commented:
Um, yes. My bad, the mixup was on my side

m1ga commented:
@gjerlow PR works fine! Just did some testing.
Could you please run `clang-format -style=file -i src/firebase/cloudmessaging/*` to format the java files and set the version number to `2.0.3`. 
I'm ready with https://github.com/hansemannn/titanium-firebase-cloud-messaging/pull/70/files and we could merge those to PRs together into a new version!

gjerlow commented:
@m1ga Sorry for the long wait, it's been some busy weeks at work. I renamed the filed name to "rounded_large_icon" to align with the [Android Notification Anatomy](https://developer.android.com/guide/topics/ui/notifiers/notifications#Templates) filed name `largeIcon`. I'm gladly changing this to something else if you have better names for it.

Added a short description to [supported data fields](https://github.com/hansemannn/titanium-firebase-cloud-messaging#data--notification-messages) and ran `clang-format`

Below is a sample of how it can look like

```
{
"title": "This is the title",
"message": "This is the message",
"big_text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do",
"icon": "https://api.adorable.io/avatars/150/test.png",
"rounded_large_icon": true,
"image": "http://via.placeholder.com/350x150",
"color": "#35A8C4",
}
```

![_20190612_122737](https://user-images.githubusercontent.com/8874119/59344509-8744c380-8d0e-11e9-9abe-df72b631a10b.JPG)

m1ga commented:
@gjerlow nice! I'll test it later today
